src/auto-evo/steps/ForceExtinction.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace AutoEvo
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Godot;
+
+    /// <summary>
+    ///   Forces extinction of worse-faring species to limit the number of steps in next auto-evo run
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     This was a hotfix to the 0.5.6 release.
+    ///     TODO: Inquire more refined solution, such as random extinction with bias for smaller populations.
+    ///   </para>
+    /// </remarks>
+    public class ForceExtinction : IRunStep
+    {
+        private readonly List<Patch> patches;
+        private readonly AutoEvoConfiguration configuration;
+
+        public ForceExtinction(List<Patch> patches, AutoEvoConfiguration configuration)
+        {
+            this.patches = patches;
+            this.configuration = configuration;
+        }
+
+        public bool CanRunConcurrently => false;
+
+        public int TotalSteps => 1;
+
+        public bool RunStep(RunResults results)
+        {
+            var populationsByPatch = results.GetPopulationsByPatch(!configuration.ProtectMigrationsFromSpeciesCap,
+                !configuration.AllowNoMutation, !configuration.ProtectNewCellsFromSpeciesCap);
```

Comment:
What's the nullability override usage here `!`? I don't think there's any other place where it is used in the codebase.

Reply:
I don't get what you mean. `!` is just a logical negation, isn't it?

Reply:
Whoops, I misread this. Anyway is it so necessary to use the negation *so* much? Can't you just assume that (at least the non-new options like AllowNoMutation), are in the worst possible option and just let tiny bit of extra work happen?

Reply:
I was thinking of https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving

Reply:
No, because then the result I get would not accurately display what the auto-evo is doing. It's not about avoiding extra work.